File: src/dtos/project_import_dtos.py
from pydantic import BaseModel

from src.dtos.edge_dtos import EdgeIncomingDto
from src.dtos.issue_dtos import IssueIncomingDto
from src.dtos.project_dtos import ProjectIncomingDto


class ProjectImportDto(BaseModel):
    projects: ProjectIncomingDto
    issues: list[IssueIncomingDto] = []
    edges: list[EdgeIncomingDto] = []

File: PrismaApi/PrismaApi.Domain/Dtos/ProjectImportDtos.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace PrismaApi.Domain.Dtos;

/// <summary>DTO for importing projects with their related entities (issues, edges, strategies).</summary>
public class ProjectImportDto
{
    [JsonPropertyName("projects")]
    public ProjectIncomingDto Projects { get; set; } = new();

    [JsonPropertyName("issues")]
    public List<IssueIncomingDto> Issues { get; set; } = new();

    [JsonPropertyName("edges")]
    public List<EdgeIncomingDto> Edges { get; set; } = new();
}
